rust/experimental/lang/src/rust.rs

```diff
@@ -38,56 +42,108 @@ pub enum StateEl {
     // generics etc
 }
 
-pub fn to_style(style: &mut Style, el: &StateEl) {
-    match *el {
-        StateEl::Comment => style.fg_color = 0xFF75715E,
-        StateEl::StrQuote => style.fg_color = 0xFF998844,
-        StateEl::CharQuote => style.fg_color = 0xFF998844,
-        StateEl::Invalid => style.fg_color = 0xFFFF0000,
-        StateEl::NumericLiteral => style.fg_color = 0xFF6644EE,
-        StateEl::CharConst => style.fg_color = 0xFF8866EE,
-        StateEl::Keyword => style.font = 1,
-        StateEl::Operator => style.fg_color = 0xFFAA2244,
-        StateEl::PrimType => style.fg_color = 0xFF44AAAA,
+impl StateEl {
+    /// Transform a [StateEl] into a [Vec] of Syntect Scope [String]s.
+    /// See [this](https://github.com/sublimehq/Packages/blob/master/Rust/Rust.sublime-syntax)
+    /// for reference.
+    pub fn as_scopes(&self) -> Scope {
+        let scope_strs = match self {
+            StateEl::Source => vec!["source.rust"],
```

Comment:
I couldn't make these either `const` or `static` since `vec![]` and `Vec::new()` aren't `const` functions. I'm pretty unfamiliar with either concept in Rust, so I'm not sure if there is some finagling I could do to get it working. I did a quick read and it seemed not really possible.

What I ended up doing was bringing in `lazy_static` and pulled out the `Vecs`-- the thought  being that it might help future refactoring. With the `to_vec()` calls, I'm still allocating as much as I was before, but I'm hoping that we can migrate the `Scope` type from `Vec<String>` to `&[&str]` or something in the future.

Reply:
right, sorry, I should've been more clear.

What I was thinking is that instead of these being `Vec<String>`s, they can be `const &[&str]`s. In fact, given that we know all of our scopes up front, it might even make sense to just give them numerical IDs; we could send `def_styles` right after launch, and then exclusively use integer ids internally?

In any case this plugin is 100% a demo so I don't think we need to worry too much.